test/unit/ldp/http/metadata/LinkRelMetadataWriter.test.ts

```diff
@@ -1,27 +1,16 @@
 import { LinkRelMetadataWriter } from '../../../../../src/ldp/http/metadata/LinkRelMetadataWriter';
 import { RepresentationMetadata } from '../../../../../src/ldp/representation/RepresentationMetadata';
-import * as util from '../../../../../src/util/HeaderUtil';
 import { LDP, RDF } from '../../../../../src/util/UriConstants';
 import { toNamedNode } from '../../../../../src/util/UriUtil';
+import { mockHttpResponse } from '../../../../util/Util';
 
 describe('A LinkRelMetadataWriter', (): void => {
   const writer = new LinkRelMetadataWriter({ [RDF.type]: 'type', dummy: 'dummy' });
-  let mock: jest.SpyInstance;
-  let addHeaderMock: jest.Mock;
-
-  beforeEach(async(): Promise<void> => {
-    addHeaderMock = jest.fn();
-    mock = jest.spyOn(util, 'addHeader').mockImplementation(addHeaderMock);
-  });
-
-  afterEach(async(): Promise<void> => {
-    mock.mockRestore();
-  });
 
   it('adds the correct link headers.', async(): Promise<void> => {
+    const response = mockHttpResponse();
     const metadata = new RepresentationMetadata({ [RDF.type]: toNamedNode(LDP.Resource), unused: 'text' });
-    await expect(writer.handle({ response: 'response' as any, metadata })).resolves.toBeUndefined();
-    expect(addHeaderMock).toHaveBeenCalledTimes(1);
-    expect(addHeaderMock).toHaveBeenLastCalledWith('response', 'link', [ `<${LDP.Resource}>; rel="type"` ]);
```

Comment:
The issue here was that we are testing implementation rather than behavior: it does not matter _how_ the class sets the headers (we should not even know), but rather that the result is correct.

Reply:
True!